### PR TITLE
MINOR: code cleanup for Kafka Streams task classes

### PR DESCRIPTION
**Pull request by mjsax**
Not functional change. Pure code cleanup

Call for review @vvcephei

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 1]
```diff
@@ -16,82 +16,85 @@
...
     private Task.State state = CREATED;
-    protected Set<TopicPartition> inputPartitions;
-    protected ProcessorTopology topology;
 
     protected final TaskId id;
+    protected final ProcessorTopology topology;
     protected final StateDirectory stateDirectory;
     protected final ProcessorStateManager stateMgr;
 
+    protected Set<TopicPartition> inputPartitions;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:40]: Align members to constructor parameter order, and group final / mutable.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 2]
```diff
@@ -16,82 +16,85 @@
...
                  final StateDirectory stateDirectory,
                  final ProcessorStateManager stateMgr,
                  final Set<TopicPartition> inputPartitions) {
         this.id = id;
-        this.stateMgr = stateMgr;
         this.topology = topology;
-        this.inputPartitions = inputPartitions;
+        this.stateMgr = stateMgr;
         this.stateDirectory = stateDirectory;
+        this.inputPartitions = inputPartitions;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:51]: Align assignment to parameter order.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java [hunk 3]
```diff
@@ -57,30 +59,31 @@
...
-                final ProcessorStateManager stateMgr,
                 final StateDirectory stateDirectory,
+                final ProcessorStateManager stateMgr,
+                final Set<TopicPartition> partitions,
+                final StreamsConfig config,
+                final InternalProcessorContext processorContext,
                 final ThreadCache cache,
-                final InternalProcessorContext processorContext) {
+                final StreamsMetricsImpl metrics) {
         super(id, topology, stateDirectory, stateMgr, partitions);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java:70]: Put "super parameter first" in constructor list

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 4]
```diff
@@ -97,13 +97,9 @@ public boolean isValidTransition(final State newState) {
         }
     }
 
-    TaskId id();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:100]: Group and order methods (compare in-line comments). -- Sub-classed inherit this ordering.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 5]
```diff
@@ -97,13 +97,9 @@ public boolean isValidTransition(final State newState) {
...
 
-    State state();
 
-    boolean isActive();
-
-    boolean isClosed();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:106]: This method is unused and removed. That is the only actual change.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 6]
```diff
@@ -116,89 +112,100 @@ public boolean isValidTransition(final State newState) {
...
+    TaskId id();
+
+    State state();
+
+    boolean isActive();
+
+    /**
+     * Updates input partitions after a rebalance
+     */
+    void updateInputPartitions(final Set<TopicPartition> topicPartitions, final Map<String, List<String>> nodeToSourceTopics);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:191]: Renamed from `update` to `updateInputPartitions`

**Comment by mjsax** [comment]
Retest this please.

**Review by vvcephei — APPROVED** [review]
My only feedback was https://github.com/apache/kafka/pull/8851#pullrequestreview-428580929

Thanks for the cleanup!

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Seem `trunk` is broker atm. Will rebase the PR when fixed.

**Review by andrewchoi5 — APPROVED** [review]
lgtm.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 7]
```diff
@@ -116,89 +112,100 @@ public boolean isValidTransition(final State newState) {
...
+
+    // IQ related methods
+
+    StateStore getStore(final String name);
+
+    /**
+     * @return the offsets of all the changelog partitions associated with this task,
+     *         indicating the current positions of the logged state stores of the task.
+     */
+    Map<TopicPartition, Long> changelogOffsets();
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:209]: just curious. the implementation of ```changelogOffsets``` (```StandbyTask``` and ```StreamTask```) return an unmodified view. By contrast, the implementation of ```inputPartitions```/```changelogPartitions``` don't wrap the return object. Is there a reason?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:209]: Just cycling back to this PR, that I will break into multiple smaller ones now...

About your question: not totally sure atm. I guess the point is, that `inputPartitions` and `changelogPartitions` don't change and we know that they are always used read-only. In contrast the changelog offset map might change, and thus we put a guard in place that the offsets are not modified from "outside".

Comment (chia7712) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:209]: @mjsax Got it. Thanks for your response!

**Pull request closed without merging**